scripts/restore-backup-to-file.sh

```diff
@@ -0,0 +1,110 @@
+    help="true"
+    break
+    ;;
+    esac
+done
+
+usage () {
+   echo "USAGE: $0 --aws-access-key <your_aws_access_key>  \ "
+   echo "          --aws-secret-access-key <your_aws_secret_access_key> \ "
+   echo "          --backup-url s3://backupbucket@ap-northeast-1/backupstore?backup=<backup_name>&volume=<volume_name> \ "
+   echo "          --target-directory /tmp/restore/volume.raw \ "
+   echo "          --output-format raw \ "
+   echo "          [--backing-file <backing_file_path>]"
+   echo "Restore a Longhorn backup to a raw image or a qcow2 image."
+   echo ""
+   echo "  -u, --backup-url             (Required) Backups URL, e.g. s3://backupbucket@us-east-1/backupstore?backup=backup-bd326da2c4414b02&volume=volumeexamplename"
+   echo "  -o, --output-file            (Required) Specify the output file name, which is saved in the /tmp/restore directory by default, e.g. volume.raw"
+   echo "  -f, --output-format          (Required) Output file foramt. Supported format: raw or qcow2."
+   echo "  -v, --version                (Required) Specify Lognhorn version to execute, e.g. v1.3.2"
+   echo "      --aws-access-key         (Optional) AWS credentials access key, e.g. AKIAIOSFODNN7EXAMPLE"
+   echo "      --aws-secret-access-key  (Optional) AWS credentials access secret key, e.g. wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"
+   echo "  -b, --backing-file           (Optional) If the backup is based on a backing file, it should provide the backing file, e.g. /tmp/bakingfile.qcow2"
+   echo "  -h, --help                   Usage message"
+}
+
+if [[ "$help" ]]; then
+    usage
+    exit 0
+fi
+
+
+# Check the required paramters exits
+if [ -z "${backup_url}" ] || [ -z "${output_file}" ] || [ -z "${output_format}" ] || [ -z "${version}" ]; then
+      missing_params="true"
+fi
+
+if [[ "${backup_url}" =~ ^[Ss]3 ]]; then
+  if [ -z "${aws_access_key}" ] || [ -z "${aws_secret_access_key}" ]; then
+      missing_params="true"
+  fi
+fi
+
+if [ "${missing_params}" ]; then
+  echo "There are REQUIRED parameters you missed!" && exit 0
+fi
+
+
+# Start restoring a backup to an image file. 
+if [[ "${backup_url}" =~ ^[Ss]3 ]]; then
+  docker run -v /tmp/restore:/tmp/restore \
+              -e AWS_ACCESS_KEY_ID="${aws_access_key}" -e AWS_SECRET_ACCESS_KEY="${aws_secret_access_key}" \
+              longhornio/longhorn-engine:"${version}" longhorn backup \
+              restore-to-file ""${backup_url}"" \
+              --output-file "/tmp/restore/${output_file}" \
+              --output-format "${output_format}" \
+              --backing-file "${backing_file}"
+else
+  docker run --cap-add SYS_ADMIN --security-opt apparmor:unconfined -v /tmp/restore:/tmp/restore \
+              longhornio/longhorn-engine:"${version}" longhorn backup \
+              restore-to-file ""${backup_url}"" \
+              --output-file "/tmp/restore/${output_file}" \
+              --output-format "${output_format}" \
+              --backing-file "${backing_file}"
+fi
```

Comment:
```suggestion
if [[ ! "${backup_url}" =~ ^[Ss]3 ]]; then
    DOCKER_CAP = "....."
    DOCKER_SEC = "....."
fi

docker run ${DOCKER_CAP} ${DOCKER_SEC} \
              -v /tmp/restore:/tmp/restore \
              longhornio/longhorn-engine:"${version}" longhorn backup \
              restore-to-file ""${backup_url}"" \
              --output-file "/tmp/restore/${output_file}" \
              --output-format "${output_format}" \
              --backing-file "${backing_file}"
fi
```

Probably can simplify the codes like this.

Reply:
Thank you. I will rewrite to simplify the code.